Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedGraphicsLayerSkia.cpp

```diff
@@ -135,31 +139,101 @@ class AcceleratedBufferPool {
     RunLoop::Timer m_releaseUnusedBuffersTimer;
 };
 
+class UnacceleratedThreadedRenderingPool {
+    WTF_MAKE_FAST_ALLOCATED();
+public:
+    static UnacceleratedThreadedRenderingPool& singleton()
+    {
+        static NeverDestroyed<UnacceleratedThreadedRenderingPool> s_pool;
+        return s_pool;
+    }
+
+    UnacceleratedThreadedRenderingPool()
+        : UnacceleratedThreadedRenderingPool(numPaintingThreads())
+    {
+    }
```

Comment:
Why don't we make the constructor just receive the painting threads count?

Reply:
I wanted to avoid the getenv() lookups everytime - we could of course also pass the numThreads in here, but then we need to cache the results of the getenv outside.

Reply:
I think we can make numPaintingThreads a static function that initializes a static variable using std::once